Create and Update Documentation

JudahZF opened the pull request:
_**This is not 100% finished, but I have created the PR to allow for review of the changes**_

### Current Changes:
- Simplify the readme file:
  - There are now 2 Readme Files, 1 with the basic information needed to install, and 1 with longer explanations of how the installation works. The difference is 50 lines, but some lines are shorter.
  - Removed VNC and Model Renaming instructions from the file. These have been replaced with links to dedicated documents.
  - The README is now markdown lint compliant
- Created base for Other Docs
  - 2 new folders under the dev folder, one called consumer, one called dev
    - Dev contains empty files for developer documentation to be placed (I will update these files before merge with some info I can quickly scrape, or a template for how info should be laid out)
    - Consumer contains instructions for using pre-built packages like the model renamer, or the VNC
- All images are now saved locally

JudahZF committed:
Formatting

JudahZF committed:
Create Base Files

JudahZF committed:
Create a Duplicate readme keeping the detailed info

JudahZF committed:
Make the Readme MarkDown Lint compliant

JudahZF committed:
Merge remote-tracking branch 'upstream/main' into Docs

JudahZF committed:
Auto stash before merge of "Docs" and "upstream/main"

JudahZF committed:
Cleanup after pulling from upstream

JudahZF committed:
Add images to MD files

JudahZF committed:
Bring detailed readme up to scratch

JudahZF committed:
Update Table of Contents

VanIseghemThomas reviewed (changes requested):
You can leave this one out. Just the normal README will be enough

(Duplicate of below, figuring out how code review works)

VanIseghemThomas commented on README Detailed.md:
```diff
@@ -0,0 +1,302 @@
+# OpenCortex
```
You can leave this one out. Just the README is enough.

JudahZF replied:
Fixed in commit

VanIseghemThomas commented on README.md:
```diff
@@ -201,18 +171,12 @@ Welcome to
-Now time for some cleanup.
-
-### Step 6: restoring the update script
+Now time for some cleanup. Make sure to be responsible now.
```
Cleanup not necesarry with the SSH shadow file, so you can leave this out. Probably something I looked over :)

JudahZF replied:
Fixed in commit

VanIseghemThomas commented on docs/consumer/Model_Renamer.md:
```diff
@@ -19,4 +19,4 @@ If you don't want to rename the files manually you can:
 Now replace the `ModelRepo.xml` file inside `/opt/neuraldsp` with the new file. Make sure this is called `ModelRepo.xml`.
 Finally reboot your QC, now you should have all models (except captures) renamed to their real names.
 
-![IMG20221218151130](https://user-images.githubusercontent.com/55881698/208303182-8554e62c-96a9-41f2-be0d-1f1f4f564506.jpg)
```
I think it's better for repo size to keep these online.

JudahZF replied:
Fixed in commit

VanIseghemThomas commented on docs/consumer/VNC.md:
```diff
@@ -1,6 +1,6 @@
 # External editor (VNC)
 
-![image](https://user-images.githubusercontent.com/55881698/214691276-bbd161bf-eb72-4f96-87ec-aa4255c75e7e.png)
```
Same here

JudahZF replied:
Fixed in commit

VanIseghemThomas reviewed (changes requested):
All seems pretty good, gonna read through it in detail when you feel like it's done, but I'm a fan of how it is structured now.

For the images it is best to keep them hosted by Github. This is easier to work with and slims down the repo size.

I would just get rid of the detailed README and maybe put all information just in the docs folder. The README can be used as a table of contents to the docs folders and to showcase some features of the project. Also maybe some links to cool stuff.

JudahZF committed:
Made Requested Changes

Removed images from repo, removed duplicate readme,

JudahZF committed:
Add Content to Developer Docs

JudahZF commented:
I have added content from the Crypto, DSP and Research channels to the Dev Docs files. I have not done the general channel as it is very large.

VanIseghemThomas committed:
Fixed small typo in Capture docs

VanIseghemThomas commented on docs/dev/Captures.md:
```diff
@@ -0,0 +1,19 @@
+# Captures
+
+[Back to developer documentaion main page](README.md)
```
Fixed this small typo.

VanIseghemThomas commented on docs/dev/README.md:
```diff
@@ -0,0 +1,50 @@
+Languages for the Project:
+
+- Python - for simple Scripting
+- Golang - for compiled binaries (like dsp)
```
Don't think Golang or Rust is included

VanIseghemThomas commented on docs/dev/README.md:
They can find this convo in the discord, going to take it out here.

JudahZF replied:
I had also intended to remove that conversation, it just made me laugh when looking at it, and I must have forgot to remove it by the end.

VanIseghemThomas committed:
Going to rewrite this in a little more detail later.

VanIseghemThomas commented on docs/dev/Updates.md:
```diff
@@ -0,0 +1,21 @@
+
+## Man In The Middle updates
+
+SSL checks are disabled everywhere (in ZenUI too, not just the updater) ... this means that with some work we would have command execution on the unit without even opening it.
```
I'm not sure if this was the case for ZenUI, and the MITM is purely on paper. Leaving that out for now.

JudahZF replied:
The in ZenUI bit was ripped straight from info in the discord, I just rewrote the start of the statement. I agree since it's currently a theory it should probably have been left out.

VanIseghemThomas committed:
Some small changes and typo fixes

Left out the hash since I'm not shure if that violates NDSP's IP. Because I'm in doubt, I'm leaving it out just to be shure.

VanIseghemThomas reviewed (approved):
Overall this looks pretty good, did some changes to it and might come back to document things in a bit more detail.

I'll ask @evilsocket to read the DSP section to verify that information is accurate or not.

JudahZF commented:
Sound great. The stuff that was left in the README was stuff that I was unsure of what to do with, just relevant info that didn't have specific categories to be sorted into.
